azurerm/resource_arm_metric_alertrule_test.go

```diff
@@ -54,6 +55,7 @@ func TestAccAzureRMMetricAlertRule_sqlDatabaseStorage(t *testing.T) {
 				Config: config,
 				Check: resource.ComposeTestCheckFunc(
 					testCheckAzureRMMetricAlertRuleExists(resourceName),
+					resource.TestCheckNoResourceAttr(resourceName, "$type"),
```

Comment:
this is looking for the field `$type` rather than the Tag `$type` - as such this won't work. Instead this wants to check for `tags.$type`

Reply:
Good catch.